server.c

```diff
@@ -687,6 +687,65 @@ server_zonestat_switch(struct nsd* nsd)
 }
 #endif /* USE_ZONE_STATS */
 
+#ifdef BIND8_STATS
+void
+server_stat_alloc(struct nsd* nsd)
+{
+	char tmpfile[256];
+	size_t sz = sizeof(struct nsdst) * nsd->child_count * 2;
+	uint8_t z = 0;
+
+	/* file name */
+	nsd->statfname = 0;
+	snprintf(tmpfile, sizeof(tmpfile), "%snsd-xfr-%d/nsd.%u.stat",
```

Comment:
Maybe use a different filename pattern? `%snsd-xfr-%d/nsd.%u.stat` may cause confusion (the `-xfr-` part)? Don't know if this is on purpose (tooling may depend on it), so just a very minor remark.

Reply:
The same directory is also used in server_zonestat_alloc server.c:540, and server_prepare_xfrd server.c:1622. For the zone stats and the difffile task mmap files. So I thought to put them in the same place. This directory is sitting in a temp directory and the zone transfers are also there. Since the directory can get the correct access permissions putting the mmapped file in there is a good idea I think.

The name of the directory stems from the use of xfrd to store zone transfers in it. Perhaps we could rename it, it is dynamically created and removed when nsd starts.

Reply:
I figured something along those lines. Feel free to keep it as-is :+1: